### PR TITLE
Simplify pawns code a bit

### PR DESCRIPTION
**Pull request by snicolet**
Simplify a bit the number of bitwise operators used to calculate the pawn evaluation in pawns.cpp

On my machine this small simplification seems to give a tiny speed-up, but I don't have the script to measure it very accurately or with confidence. 

No functional change.

### CODE REVIEW

**Comment by zamar** [comment]
approved.

It's for understandable this way (At least for me)

File: src/pawns.cpp [hunk 1]
```diff
@@ -193,7 +191,7 @@ namespace {
             score -= Backward[opposed][f];
 
         if (connected)
-            score += Connected[opposed][phalanx][more_than_one(supported)][relative_rank(Us, s)];
+            score += Connected[opposed][!!phalanx][more_than_one(supported)][relative_rank(Us, s)];
```
Comment (lucasart) [inline src/pawns.cpp]: probably a matter of personal preference, but I find `phalanx != 0` more readable than `!!phalanx`.

Comment (mstembera) [inline src/pawns.cpp]: How about ```(bool)phalanx``` ?

Comment (Nazine) [inline src/pawns.cpp]: phalanx !=0 is ten times more readable than any of your headless suggestions.

Comment (zamar) [inline src/pawns.cpp]: !! is traditional SF coding style. I see nothing wrong with that.

At first, it looks strange, but when you get used to the notation it's very compact...

**Pull request closed without merging**